packages/design-system/src/new-ui/DataTable/DataTablePagination.tsx

```diff
@@ -0,0 +1,48 @@
+import { Table } from "@tanstack/react-table";
+import { Button } from "../Button";
+import { Icons } from "../Icons";
+
+interface DataTablePaginationProps<TData> {
+  table: Table<TData>;
+}
+
+export function DataTablePagination<TData>({
+  table,
+}: DataTablePaginationProps<TData>) {
+  return (
+    <div className="flex items-center justify-end py-8">
+      <Button
+        className="h-10 gap-x-3 !rounded-r-none rounded-l-[8px]"
+        variant="secondaryGrey"
+        size="sm"
+        onClick={() => table.previousPage()}
+        disabled={!table.getCanPreviousPage()}
+      >
+        <Icons.ArrowNarrowLeft className="h-4 w-4 stroke-semantic-fg-secondary" />
+        Previous
+      </Button>
+      {table.getPageOptions().map((e, index) => (
+        <Button
+          className="h-10 w-10 !rounded-none"
+          variant="secondaryGrey"
+          size="sm"
+          onClick={() => table.setPageIndex(index)}
+          key={`table-page-button-${index}`}
+          disabled={index === table.options.state.pagination?.pageIndex}
+        >
+          {index + 1}
+        </Button>
+      ))}
+      <Button
+        className="h-10 gap-x-3 !rounded-l-none rounded-r-[8px]"
+        variant="secondaryGrey"
+        size="sm"
+        onClick={() => table.nextPage()}
+        disabled={!table.getCanNextPage()}
+      >
+        Next
+        <Icons.ArrowNarrowRight className="h-4 w-4 stroke-semantic-fg-secondary" />
+      </Button>
+    </div>
```

Comment:
@iamnamananand996 The style is not correct. 

- In the design doc, the pagination button are not fixed width nor height but a Hug component with padding. You can overwrite the button's padding. 
- We have `rounded-sm` for `rounded-[8px]`. Please do remember to find whether the style has its token before using arbitrary value